src/ophyd_async/core/signal.py

```diff
@@ -63,29 +70,95 @@ def __init__(
     ) -> None:
         self._timeout = timeout
         self._initial_backend = self._backend = backend
+
         super().__init__(name)
 
+    def children(self) -> Iterator[Tuple[str, "Device"]]:
+        yield from ()
+
     async def connect(
         self,
         mock=False,
         timeout=DEFAULT_TIMEOUT,
         force_reconnect: bool = False,
         backend: Optional[SignalBackend[T]] = None,
     ):
-        if backend:
-            if self._initial_backend and backend is not self._initial_backend:
-                raise ValueError(
-                    "Backend at connection different from initialised one."
-                )
-            self._backend = backend
-        if mock and not isinstance(self._backend, MockSignalBackend):
-            # Using a soft backend, look to the initial value
-            self._backend = MockSignalBackend(initial_backend=self._backend)
-
-        if self._backend is None:
-            raise RuntimeError("`connect` called on signal without backend")
-        self.log.debug(f"Connecting to {self.source}")
-        await self._backend.connect(timeout=timeout)
+        """
+        Connect self with a timeout
+
+                          +---------------------+
+                          |     Start:          |
+                          |     connect()       |
+                          +---------------------+
+                                    |
+                                    V
+                  +-----------------------------------+
+                  |   Check if previous mock state    |
+                  |   exists and differs from current |
+                  +-----------------------------------+
+                          | Yes                No
+                          |                      |
+                          V                      V
+              +---------------------+  +-----------------------+
+              |  Fail with error:   |  |   Proceed with        |
+              |  "Inconsistent use  |  |   connection setup:   |
+              |  of mock parameter" |  |   - Check backend     |
+              +---------------------+  |   - Initiate connect  |
+                                      +-----------------------+
+                                               |
+                                               V
+                                 +---------------------------+
+                                 |  Continue operations      |
+                                 |  based on backend and     |
+                                 |  mock settings            |
+                                 +---------------------------+
+
+        Exclude conditions:
+            - Switching between mock=True and mock=False (and vice versa)
+                across successive connect calls.
+            - Passing a different backend than the one initially set
```

Comment:
@evalott100 I would like to use a clear chart like this as a reference point. could you tell me 
1) if it's not wrong
2) if it's not missing anything crucial?

Reply:
I'd argue we don't need a chart here - and starting to regularly put them in docstrings would be a huge mistake.

All this information could be encapsulated in descriptions of the args - i.e a small sentence `"The value can/cannot be changed between connects.`" on the description for `mock`.

As for what it is missing within the docstring I'd say pretty much just the "mock" explanation above, and specifying the behavior of the connect asyncio task (i.e if `connect` while the connect task isn't active).

Reply:
deleted the chart